cinder/scheduler/filters/sap_large_volume_filter.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020 SAP SE
+#
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from cinder.scheduler import filters
+
+
+LOG = logging.getLogger(__name__)
+CONF = cfg.CONF
+
+
+class SAPLargeVolumeFilter(filters.BaseBackendFilter):
+    """Filter out volumes from landing on vvol datastores for > 2TB"""
+
+    def backend_passes(self, backend_state, filter_properties):
+        host = backend_state.host
+        req_spec = filter_properties.get('request_spec')
+        volume_size = req_spec["volume"]["size"]
+
+        if 'vvol' in host.lower() and volume_size > 2000:
+            LOG.warning("Cannot allow volumes larger than 2000 on vVol.")
```

Comment:
Does this have to be a warning? Don't warnings end up in sentry, too? I see this as an `info` at max, as the filter is working as expected.

I'd also add a unit to the 2000 to make sure everybody can understand the log line.

Reply:
done